crates/services/sync/src/import.rs

```diff
@@ -174,18 +175,32 @@ where
             ..
         } = &self;
 
-        get_headers_buffered(range.clone(), params, p2p.clone())
-        .map({
-            let p2p = p2p.clone();
-            let consensus_port = consensus.clone();
-            move |result| {
-                Self::get_block_for_header(result, p2p.clone(), consensus_port.clone())
-            }
-            .instrument(tracing::debug_span!("consensus_and_transactions"))
-            .in_current_span()
+        let shutdown_signal = shutdown.clone();
+        let (shutdown_guard, mut shutdown_guard_recv) =
+            tokio::sync::mpsc::channel::<()>(1);
+        let block_stream =
+            get_block_stream(range.clone(), params, p2p.clone(), consensus.clone());
+        let guard_stream = futures::stream::repeat_with(move || {
+            (shutdown_guard.clone(), shutdown_signal.clone())
+        });
```

Comment:
It seems overkill because below, you will clone it directly during spawning task.

Reply:
I need to clone it either way: either like this, or in `stream.map` like:

```
let block_stream = get_block_stream(range.clone(), params, p2p.clone(), consensus.clone());
let result = block_stream
.map(move |stream_block_batch| {
    let shutdown_guard = shutdown_guard.clone();
    let shutdown_signal = shutdown_signal.clone();
    tokio::spawn(async move {
        // Hold a shutdown sender for the lifetime of the spawned task
        let _shutdown_guard = shutdown_guard.clone();
        let mut shutdown_signal = shutdown_signal.clone();
...
```

Do you have a preference?

Reply:
I think cloning inside the map is fine, let's go with that